drivers/adc/adc_mcux_adc16.c

```diff
@@ -124,7 +124,8 @@ static int mcux_adc16_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
-	if (channel_cfg->reference != ADC_REF_INTERNAL) {
+	if (channel_cfg->reference != ADC_REF_INTERNAL &&
+		channel_cfg->reference != ADC_REF_VDD_1) {
```

Comment:
I think what this was saying before is essentially that this driver doesn't support configuring the reference, so I don't think this change does anything

Reply:
The change has effect in context of samples/drivers/adc/adc_dt sample. Function adc_raw_to_millivolts_dt is called in the test. If reference is ADC_REF_INTERNAL, it tries to get reference voltage from driver api->ref_internal, but the driver adc_mcux_adc16.c does not have the api->ref_internal implemented.
Setting the reference to ADC_REF_VDD_1 (for devices which use VDD as reference) enables ref voltage configuration in Devicetree and getting correct value from adc_raw_to_millivolts_dt. It's the reason why ADC_REF_VDD_1 should be allowed in the driver.

Reply:
which platform are you talking about? Also, this seems like a bug with the use of ADC_REF_INTERNAL in the driver being incorrect

Reply:
I have removed the check and updated commit message.